custom_components/bms_ble/plugins/ant_legacy_bms.py

```diff
@@ -0,0 +1,186 @@
+                "manufacturer_id": 1623,
+                "connectable": True,
+            }
+        ]
+
+    @staticmethod
+    def device_info() -> dict[str, str]:
+        """Return device information for the battery management system."""
+        return {"manufacturer": "ANT", "model": "Smart BMS"}
+
+    @staticmethod
+    def uuid_services() -> list[str]:
+        """Return list of 128-bit UUIDs of services required by BMS."""
+        return [normalize_uuid_str("ffe0")]  # change service UUID here!
+
+    @staticmethod
+    def uuid_rx() -> str:
+        """Return 16-bit UUID of characteristic that provides notification/read property."""
+        return "ffe1"
+
+    @staticmethod
+    def uuid_tx() -> str:
+        """Return 16-bit UUID of characteristic that provides write property."""
+        return "ffe1"
+
+    @staticmethod
+    def _calc_values() -> frozenset[BMSvalue]:
+        return frozenset(
+            ("battery_charging", "cycle_capacity", "power")
+        )  # calculate further values from BMS provided set ones
+
+    def _notification_handler(
+        self, _sender: BleakGATTCharacteristic, data: bytearray
+    ) -> None:
+        """Handle the RX characteristics notify event (new data arrives)."""
+
+        self._log.debug("RX BLE data: %s", data)
+
+        if data.startswith(BMS._RX_HEADER_RSP_STAT):
+            self._data = _data = data
+            self._exp_len = BMS._RSP_STAT_LEN
+        else:
+            _data = self._data
+            if not _data:
+                self._log.debug("invalid Start of Frame")
+                return
+            _data += data
+
+        _data_len = len(_data)
+        if _data_len < self._exp_len:
+            return
+
+        if _data_len > self._exp_len:
+            self._log.debug("invalid length %d > %d", _data_len, self._exp_len)
+            self._data.clear()
+            return
+
+        if (crc := crc_sum(_data[4:-2], 2)) != (
+            remote_crc := _data[-1] | (_data[-2] << 8)
+        ):
+            self._log.debug("invalid checksum 0x%X != 0x%X", crc, remote_crc)
+            self._data.clear()
+            return
+
+        self._data_final = _data.copy()
+        _data.clear()
+        self._data_event.set()
+
+    @staticmethod
+    def _cmd(cmd: CMD, adr: ADR, value: int = 0x0000) -> bytes:
+        """Assemble a ANT BMS command."""
+        value_hi = (value >> 8) & 0xFF
+        value_lo = value & 0xFF
+        return bytes(
+            (cmd, cmd, adr, value_hi, value_lo, (adr + value_hi + value_lo) & 0xFF)
+        )
+
+    async def _async_update(self) -> BMSsample:
+        """Update battery status information."""
+        await self._await_reply(BMS._cmd(BMS.CMD.GET, BMS.ADR.STATUS))
+
+        _data = self._data_final
+
+        result = BMS._decode_data(
+            BMS._FIELDS,
+            _data,
+            byteorder="big",
+            offset=0,
+        )
+        result["cell_voltages"] = [
```

Comment:
There is a function
https://github.com/patman15/BMS_BLE-HA/blob/31e27c446e669758d99451e601eb270459914365/custom_components/bms_ble/plugins/basebms.py#L514
available to do this common operation.